### PR TITLE
[3] Replace basic auth sign in with OpenID connect

### PR DESCRIPTION
**Pull request by tahb**
While we wait to be given access to DfE Sign in we can bridge the technical gap closer from basic auth and to open-id connect - the same authentication mechanism. This pull request removes the second layer of basic auth provisioned for hiring staff but continues to wrap the whole application in basic auth when not in production.

Being able to add and remove permissions is yet to be resolved and will come in the next piece of work: https://trello.com/c/gooROCIO

We still have a easy way to override the school urn for a logged in user to facilitate testing until we have better permission management in place. A default testing user has been created and pinned to Slack.

1. Landing screen
screen shot 2018-04-05 at 15 53 51

2. Click sign in and get taken to Microsoft
screen shot 2018-04-05 at 15 56 13

3. Provide credentials and arrive back
screen shot 2018-04-05 at 15 56 44

4. Sign out
screen shot 2018-04-05 at 15 56 56

Microsoft Azure can be styled later following this guide: https://docs.microsoft.com/en-us/azure/active-directory/customize-branding

### CODE REVIEW

**Comment by tahb** [comment]
[enable schools to log out](https://trello.com/c/OWjT5NV8/2-enable-schools-to-log-out)

**Comment by tahb** [comment]
[Spin up DfE sign-on](https://trello.com/c/VhnJVvJN/3-spin-up-dfe-sign-on)

**Comment by tahb** [comment]
Branding has been set up on the 30 day free trial that should more than cover us in the transition period.

screen shot 2018-04-06 at 12 27 57

File: app/controllers/hiring_staff/sessions_controller.rb [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+  end
+
+  def failure
+    Rollbar.log('error', 'Sign in provider returned a failure')
+    render html: I18n.t('errors.sign_in.failure')
+  end
+
+  def destroy
+    session.destroy
+    redirect_to root_path, notice: "You've been signed out."
```
Comment (despo) [inline app/controllers/hiring_staff/sessions_controller.rb]: text should be extracted to locale

**Pull request merged**